### PR TITLE
New consequence terms and severity order

### PR DESCRIPTION
**Pull request by leexgh**
Fix: https://github.com/genome-nexus/genome-nexus/issues/718
Update consequence terms and variant classification

### CODE REVIEW

**Comment by leexgh** [comment]
Need variant classification for:
- feature_elongation
- feature_truncation
- coding_transcript_variant
- sequence_variant

File: component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java [hunk 1]
```diff
@@ -96,49 +97,56 @@ private static Map<String, Integer> initEffectPriority()
...
+        effectPriority.put("tf_binding_site_variant", 20); // A sequence variant located within a transcription factor binding site
+        effectPriority.put("regulatory_region_ablation", 20); // A feature ablation whereby the deleted region includes a regulatory region
+        effectPriority.put("regulatory_region_amplification", 20); // A feature amplification of a region containing a regulatory region
+        effectPriority.put("regulatory_region_variant", 20); // A sequence variant located within a regulatory region
+        effectPriority.put("regulatory_region", 20); // snpEff-specific effect that should really be regulatory_region_variant
+        effectPriority.put("intergenic_variant", 21); // A sequence variant located in the intergenic region, between genes
+        effectPriority.put("intergenic_region", 21); // snpEff-specific effect that should really be intergenic_variant
+        effectPriority.put("sequence_variant", 21); // A sequence_variant is a non exact copy of a sequence_feature or genome exhibiting one or more sequence_alteration
```
Comment (leexgh) [inline component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java:148]: this is new

File: component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java [hunk 2]
```diff
@@ -96,49 +97,56 @@ private static Map<String, Integer> initEffectPriority()
...
+        effectPriority.put("conservative_missense_variant", 9); // A sequence variant whereby at least one base of a codon is changed resulting in a codon that encodes for a different but similar amino acid. These variants may or may not be deleterious
+        effectPriority.put("rare_amino_acid_variant", 9); // A sequence variant whereby at least one base of a codon encoding a rare amino acid is changed, resulting in a different encoded amino acid
+        // low
+        effectPriority.put("splice_donor_5th_base_variant", 10); // A sequence variant that causes a change at the 5th base pair after the start of the intron in the orientation of the transcript
+        effectPriority.put("splice_region_variant", 10); // A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron
+        effectPriority.put("splice_donor_region_variant", 10); // A sequence variant that falls in the region between the 3rd and 6th base after splice junction (5' end of intron)
```
Comment (leexgh) [inline component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java:115]: new

File: component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java [hunk 3]
```diff
@@ -96,49 +97,56 @@ private static Map<String, Integer> initEffectPriority()
...
+        effectPriority.put("splice_polypyrimidine_tract_variant", 10); // A sequence variant that falls in the polypyrimidine tract at 3' end of intron between 17 and 3 bases from the end (acceptor -3 to acceptor -17)
```
Comment (leexgh) [inline component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java:116]: new

**Review by inodb — APPROVED** [review]
LGTM - thank you!

File: component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java [hunk 4]
```diff
@@ -96,49 +97,56 @@ private static Map<String, Integer> initEffectPriority()
...
+        effectPriority.put("inframe_deletion", 7); // An inframe non synonymous variant that deletes bases from the coding sequence
+        effectPriority.put("protein_altering_variant", 8); // A sequence variant which is predicted to change the protein encoded in the coding sequence
+        effectPriority.put("missense_variant", 9); // A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved
+        effectPriority.put("conservative_missense_variant", 9); // A sequence variant whereby at least one base of a codon is changed resulting in a codon that encodes for a different but similar amino acid. These variants may or may not be deleterious
+        effectPriority.put("rare_amino_acid_variant", 9); // A sequence variant whereby at least one base of a codon encoding a rare amino acid is changed, resulting in a different encoded amino acid
+        // low
+        effectPriority.put("splice_donor_5th_base_variant", 10); // A sequence variant that causes a change at the 5th base pair after the start of the intron in the orientation of the transcript
```
Comment (leexgh) [inline component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java:113]: new

File: component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java [hunk 5]
```diff
@@ -96,49 +97,56 @@ private static Map<String, Integer> initEffectPriority()
...
+        effectPriority.put("non_coding_exon_variant", 15); // A sequence variant that changes non-coding exon sequence
+        effectPriority.put("non_coding_transcript_exon_variant", 15); // snpEff-specific synonym for non_coding_exon_variant
+        effectPriority.put("non_coding_transcript_variant", 16); // A transcript variant of a non coding RNA gene
+        effectPriority.put("nc_transcript_variant", 16); // A transcript variant of a non coding RNA gene (older alias for non_coding_transcript_variant)
+        effectPriority.put("intron_variant", 16); // A transcript variant occurring within an intron
+        effectPriority.put("intragenic_variant", 16); // A variant that occurs within a gene but falls outside of all transcript features. This occurs when alternate transcripts of a gene do not share overlapping sequence
+        effectPriority.put("intragenic", 16); // snpEff-specific synonym of intragenic_variant
+        effectPriority.put("nmd_transcript_variant", 17); // A variant in a transcript that is the target of NMD
+        effectPriority.put("coding_transcript_variant", 18); // A transcript variant of a protein coding gene
```
Comment (leexgh) [inline component/src/main/java/org/cbioportal/genome_nexus/component/annotation/TranscriptConsequencePrioritizer.java:136]: new

**Pull request merged**